crates/noirc_evaluator/src/ssa/ir/instruction.rs

```diff
@@ -756,6 +756,16 @@ impl Binary {
 
                 let lhs = truncate(lhs.try_into_u128()?, *bit_size);
                 let rhs = truncate(rhs.try_into_u128()?, *bit_size);
+
+                // The divisor is being truncated into the type of the operand, which can potentially
+                // lead to the rhs being zero.
+                // If the rhs of a division is zero, attempting to evaluate the divison will cause a compiler panic.
+                // Thus, we do not evaluate the division in this method, as we want to avoid triggering a panic,
+                // and the operation should be handled by ACIR generation.
+                if matches!(self.operator, BinaryOp::Div) && rhs == 0 {
+                    return None;
+                }
```

Comment:
I don't think this is the right solution to the problem. The problem is that code like `1 >> 8` is getting turned into `1 / (2^8 mod 256)` for u8, and checking for zero here would cause ACIR-gen to fail due to https://github.com/noir-lang/noir/pull/2475, when the correct behavior should be returning zero.

I think the right solution here would be to fix `self.pow` in the ssa gen context to avoid the `mod 256` after the pow.

Reply:
It will not fail during ACIR gen as the rhs is in fact `2` not `0` (I added a test under `execution_success/bit_shifts_comptime/src/main.nr` to show that the code still passes). During `eval_constants` we have the constant `2` but it is being truncated to `0` as the operand type is `u1`.

Reply:
I don't see that any truncations on the rhs of the bit shift (https://github.com/noir-lang/noir/blob/b55f3998258e9c82a0312fc138e3c6b5cf1a7c74/crates/noirc_evaluator/src/ssa/ssa_gen/context.rs#L251) unless you are looking somewhere else?

Reply:
For this code:
```
	let b: u8 = 1 >> 8;
	assert(b == 0);
```
you will have this SSA:
```
After Dead Instruction Elimination:
acir fn main f2 {
  b0(v0: u64):
    v8 = div u8 1, Field 2⁸
    constrain v8 == u8 0
    return 
}
```

Reply:
Hmm ok. I thought pow may have been inserting them. It seems a bit odd that acir-gen is fine with u8 values being larger than a u8 can hold but it works for our usecase here.

Reply:
Yeah I think it is because the result of `pow` is a field. Perhaps it is worth to explore being a bit more restrictive and inserting a truncate on the result of `pow`.